static/js/narrow.js

```diff
@@ -535,6 +551,10 @@ function pick_empty_narrow_banner() {
             // You have no private messages.
             return $("#empty_narrow_all_private_message");
         }
+    } else if ((first_operator === "stream") && !stream_data.is_subscribed(first_operand)
+                                             && sub.invite_only) {
+        // You are narrowed to a invite only stream to which you aren't subscribed
+        return $("#non_existing_stream_message");
```

Comment:
Yeah. user won't be able to see anything of invite_only stream. But, as you mentioned earlier in this discussion, narrowing to a `invite_only` should raise "Stream does not exist" error message.